cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -0,0 +1,169 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+)
+
+// PoolOperator is the name of the tool that makes pool-related operations.
+const (
+	PoolOperator = "zpool"
+)
+
+var RunnerVar util.Runner
+
+// ImportPool imports cStor pool if already present.
+func ImportPool(cStorPool *apis.CStorPool) error {
+	importAttr := importPoolBuilder(cStorPool)
+
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, importAttr...)
+	if err != nil {
+		glog.Error("Unable to import pool:", err.Error(), string(stdoutStderr))
+		return err
+	}
+
+	glog.Info("Importing Pool Successful")
+	return nil
+}
+
+// importPoolBuilder is to build pool import command.
+func importPoolBuilder(cStorPool *apis.CStorPool) []string {
+	// populate pool import attributes.
+	var importAttr []string
+	importAttr = append(importAttr, "import")
+	if cStorPool.Spec.PoolSpec.CacheFile != "" {
+		importAttr = append(importAttr, "-c", cStorPool.Spec.PoolSpec.CacheFile,
+			"-o", cStorPool.Spec.PoolSpec.CacheFile)
+	}
+
+	importAttr = append(importAttr, "cstor-"+string(cStorPool.ObjectMeta.UID))
+
+	return importAttr
+}
+
+// CreatePool creates a new cStor pool.
+func CreatePool(cStorPool *apis.CStorPool) error {
+	createAttr := createPoolBuilder(cStorPool)
+	glog.V(4).Info("createAttr : ", createAttr)
+
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, createAttr...)
+	if err != nil {
+		glog.Error("Unable to create pool:", err.Error(), string(stdoutStderr))
+		return err
+	}
+
+	glog.Info("Creating Pool Successful")
+	return nil
+}
+
+// createPoolBuilder is to build create pool command.
+func createPoolBuilder(cStorPool *apis.CStorPool) []string {
+	// populate pool creation attributes.
+	var createAttr []string
+	createAttr = append(createAttr, "create")
+	if cStorPool.Spec.PoolSpec.CacheFile != "" {
+		cachefile := "cachefile=" + cStorPool.Spec.PoolSpec.CacheFile
+		createAttr = append(createAttr, "-o", cachefile)
+	}
+
+	openebsPoolname := "io.openebs:poolname=" + cStorPool.Name
+	createAttr = append(createAttr, "-O", openebsPoolname)
+
+	poolNameUID := "cstor-" + string(cStorPool.ObjectMeta.UID)
+	createAttr = append(createAttr, poolNameUID)
+
+	// To generate mirror disk0 disk1 mirror disk2 disk3 format.
+	for i, disk := range cStorPool.Spec.Disks.DiskList {
+		if cStorPool.Spec.PoolSpec.PoolType == "mirror" && i%3 == 0 {
+			createAttr = append(createAttr, "mirror")
+		}
+		createAttr = append(createAttr, disk)
+	}
+
+	return createAttr
+
+}
```

Comment:
Spare disk addition should be present as part of mirror pool creation.

Reply:
Can you pls explain? Do we need to handle update CRs?